lib/util.h

```diff
@@ -94,20 +95,18 @@ extern const char CYRUS_VERSION[];
 #endif
 
 #define BIT32_MAX 4294967295U
+#define BIT64_MAX 18446744073709551615UL
 
-#if UINT_MAX == BIT32_MAX
-typedef unsigned int bit32;
-#elif ULONG_MAX == BIT32_MAX
-typedef unsigned long bit32;
-#elif USHRT_MAX == BIT32_MAX
-typedef unsigned short bit32;
-#else
-#error dont know what to use for bit32
-#endif
+#define BIT64_FMT          "%016" PRIx64
+#define UINT64_FMT         "%" PRIu64
```

Comment:
https://github.com/cyrusimap/cyrus-imapd/pull/3748 has removed `INT_FMT` ⇔ "%lld"/ "%ld" in favour of `"%" PRId64`.

I think that using PRIu64 instead of UINT64_FMT in the code will allow people, which do not have deep understanding about the Cyrus internals, to understand the code.  For this reason I do not think that adding `UINT64_FMT` is a good idea.  Using `UINT64_FMT` instead of `"%" PRIu64` does not safe any bytes.